src/tqec/detectors/gate.py

```diff
@@ -4,18 +4,22 @@
 import cirq
 import stim
 from tqec.detectors.measurement_map import CircuitMeasurementMap
-from tqec.exceptions import MeasurementAppliedOnMultipleQubitsException, QubitTypeException
+from tqec.exceptions import (
+    MeasurementAppliedOnMultipleQubitsException,
+    QubitTypeException,
+    TQECException,
+)
 
 
-class GlobalMeasurementLookbackMissingException(Exception):
+class GlobalMeasurementLookbackMissingException(TQECException):
```

Comment:
Some exceptions seem to be a simple wrapper around `TQECException` and I wonder whether we need to define those too-detailed exceptions. For example, `GlobalMeasurementLookbackMissingException()` can be replaced with the `TQECException("Global measurement lookback offsets have not been computed...")`. And this is also what `qiskit` did for some of its exceptions with `QiskitError("...")`.

Or perhaps we need the perspectives of others on this issue.